src/pages/Governance/hooks/useGetCoinSupplyLimit.tsx

```diff
@@ -0,0 +1,43 @@
+import {useGetAccountResources} from "../../../api/hooks/useGetAccountResources";
+
+interface CoinInfo {
+  decimals: number;
+  name: string;
+  supply: {
+    vec: [
+      {
+        aggregator: {
+          vec: [
+            {
+              handle: string;
+              key: string;
+              limit: string;
+            },
+          ];
+        };
+        integer: {
+          vec: [];
+        };
+      },
+    ];
+  };
+  symbol: string;
+}
+
+export function useGetCoinSupplyLimit(): string | null {
```

Comment:
is it possible that this function return `null`? i.e that `0x1` will not have the `0x1::coin::CoinInfo<0x1::aptos_coin::AptosCoin>` resource? if not, we might want to utilize `useEffect` (and potentially `useState`) to update the returned data

Reply:
I think right now we don't have much to worry about it since unlike `useGetProposal`, here we don't have any async functions to wait.

Reply:
correct, but the return type is `string | null` and I wonder if there is a real option that coin supply will return `null` or we just use the `null` return type to support the condition checking

Reply:
Oh I got it. So here just use `null` as condition checking.